geowebcache/wms/src/main/java/org/geowebcache/config/GetCapabilitiesConfiguration.java

```diff
@@ -270,6 +279,23 @@ private String getWMSUrl(WebMapServer wms) {
                     paramFilters.add(new NaiveWMSDimensionFilter(dimension, dimExtent));
                 }
 
+                if (this.cachedParameters != null && this.cachedParameters.length() != 0) {
+                    String[] cparams = this.cachedParameters.split("&");
+                    for (String cp : cparams) {
+                        if (cp.length() > 0) {
+                            String[] split = cp.split("=", 2);
+                            String key = split[0];
+                            String defaultValue = split[1];
+                            log.info(key + " and " + defaultValue);
+                            if (key.length() > 0) {
+                                paramFilters.add(
+                                    new FreeStringParameterFilter(key, defaultValue)
```

Comment:
Completely free parameters are potentially dangerous. I try to discourage people from using them so having the GetCapabilitiesConfiguration using them behind the scenes strikes me as a bad idea.  I suppose we are already in that boat with the `NaiveWMSDimensionFilter` though.